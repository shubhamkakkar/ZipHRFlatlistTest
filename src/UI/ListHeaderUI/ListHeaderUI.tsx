import React from 'react';
import { View, Text } from 'react-native';
import styles from './styleSheetListHeaderUI';

type TListHeaderUI = {
    title: string
}

export default function ListHeaderUI({ title }: TListHeaderUI) {
  return (
    <View style={styles.container}>
      <View style={styles.listHeaderTitleContainer}>
        <Text style={styles.title}>{title}</Text>
      </View>
    </View>
  );
}
