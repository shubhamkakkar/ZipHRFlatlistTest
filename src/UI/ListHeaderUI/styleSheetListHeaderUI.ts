import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 10,
    marginTop: 5,
  },
  listHeaderTitleContainer: {
    borderRadius: 20,
    backgroundColor: '#E366A720',
    paddingVertical: 10,
    paddingHorizontal: 20,
  },
  title: {
    color: '#E366A7',
    fontWeight: 'bold',
  },
});

export default styles;
