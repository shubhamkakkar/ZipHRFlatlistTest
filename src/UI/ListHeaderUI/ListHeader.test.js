import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ListHeaderUI from './ListHeaderUI';

it('ListHeaderUI matches snapshot - props: {{ title: "" }}', () => {
  const tree = renderer.create(<ListHeaderUI {...{ title: '' }} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ListHeaderUI matches snapshot - props: {{ title: "title" }}', () => {
  const tree = renderer.create(<ListHeaderUI {...{ title: 'title' }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
