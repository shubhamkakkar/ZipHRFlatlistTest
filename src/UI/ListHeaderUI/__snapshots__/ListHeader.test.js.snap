// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeaderUI matches snapshot - props: {{ title: "" }} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 5,
      "marginVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#3e0f7910",
        "borderRadius": 20,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3e0f79",
          "fontWeight": "bold",
        }
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`ListHeaderUI matches snapshot - props: {{ title: "title" }} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 5,
      "marginVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#3e0f7910",
        "borderRadius": 20,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3e0f79",
          "fontWeight": "bold",
        }
      }
    >
      title
    </Text>
  </View>
</View>
`;
